src/types.ts

```diff
@@ -325,11 +333,113 @@ export class TypedValues {
     }
 }
 
+const DECIMAL_REGEX = /^-?\d+(\.\d+)?/;
+const UUID_REGEX = /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/;
+
 const parseLong = (input: string|number): Long|number => {
    const long = typeof input === 'string' ? Long.fromString(input) : Long.fromNumber(input);
    return long.high ? long : long.low;
 };
 
+function toLong(value: Long | number): Long {
+    if (typeof value === 'number') {
+        return Long.fromNumber(value);
+    }
+    return value;
+}
+
+function trimFractionalValue(value: string) {
+    if (value.indexOf('.') >= 0) {
+        value = value.replace(/0.*$/, '');
```

Comment:
That actually means 'cut zero digit and any amount of characters till the end of the string', right? Or was the intention different?

Reply:
Yes, it does.